### PR TITLE
fix: Anchor an arrow to its slide instead of its containing block

### PR DESCRIPTION
**Pull request by whitphx**
Towards #86.

## The bug

An arrow's `<svg>` is `position: absolute`, so the browser lays it out at the top-left corner of its **containing block** — the nearest positioned or transformed ancestor — which is the slide only when nothing between the arrow and the slide happens to be one. The coordinates `x1`/`y1`, `from="(x, y)"` and percentages are written in are slide coordinates, but they were handed to that SVG unchanged, so an arrow inside something like `<div class="relative">`, a `v-drag` container, or any transformed wrapper drew them offset by that wrapper's own position. The scale had the same shape of problem: it came from Slidev's slide scale, so anything scaling the arrow between the slide and itself (a `transform: scale()` wrapper, or a slide with a `zoom` frontmatter) stretched the result.

Which elements establish a containing block is also a place where engines differ (`filter`, `will-change`, `contain`, view transitions, …), which fits #86: the same deck coming out right in one browser and wrong in another, and a reporter working around it by adding `position: relative`.

Reproduced on Chromium 141 / Linux with a slide holding

```html
<div style="position: relative; left: 200px; top: 150px">
  <FancyArrow x1="500" y1="100" x2="100" y2="300" static />
</div>
```

which drew the arrow 200px right and 150px down from where it was asked to be.

## The fix

`components/arrow-frame.ts` is a new pure module that measures the mapping between the three coordinate systems an arrow deals with — screen (what `getBoundingClientRect()` returns), the SVG's user units (what the arrow is drawn in), and slide coordinates (what the props are written in) — and `position.ts` converts through it:

- The slide's origin is measured against the SVG's, so slide coordinates land on the slide wherever the SVG got anchored.
- The scale comes from the SVG's own box. It is laid out as 1px × 1px, so its measured size *is* how much screen space one user unit takes, whatever the slide and everything under it do to the arrow. Slidev's `$scale` stays as the fallback for when the SVG has no size to measure.
- Snapped endpoints keep being measured against the SVG itself, which was already right, and now use the measured scale instead of the slide's.

An arrow with no slide around it (`.slidev-page` missing) keeps its coordinates as they are, which is what it did before.

## Verification

- `pnpm test` (117 tests, including the new `components/arrow-frame.test.ts`), `pnpm lint`, `pnpm format --check`, `pnpm build`.
- Rendered the demo deck in headless Chromium before and after: existing slides are unchanged, and the repro above now draws where it is told. On a `zoom: 0.7` slide, an arrow from `(0, 0)` to `(980, 552)` now spans the slide corner to corner; before it stopped at 70% of it. An arrow inside a `transform: scale(1.5)` wrapper now snaps to its target instead of overshooting by half.

I could not reproduce the exact deck from #86 (the screenshots there are not reachable from this environment), so this fixes a definite cause of that class of symptom rather than a confirmed one. Worth asking the reporter to re-check once released.

---
_Generated by [Claude Code](https://claude.ai/code/session_01GtH2tR7qZzwTwGo9iqxCM9)_

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected arrow positioning and scaling when arrows appear inside positioned or transformed elements, such as draggable containers.
  * Fixed arrow alignment on slides using zoom settings.
  * Ensured arrows remain anchored to their slide rather than shifting based on surrounding elements.

* **Tests**
  * Added coverage for coordinate conversion, scaling, positioning, and edge cases.

### CODE REVIEW

File: components/position.ts [hunk 1]
```diff
@@ -76,77 +83,124 @@ export interface BoxPosition {
...
-    }
-    if ("x" in endpointRef.value) {
-      // Endpoint is of type Position
-      // so we don't need to update point in this method
-      // as it's done in the watch below.
+  const position = ref<ResolvedPosition | undefined>(undefined);
+  const updatePosition = () => {
+    const endpoint = endpointRef.value;
+    if (endpoint == null) {
       return;
```
[the remark this answers is not in the document]
Comment (whitphx) [reply components/position.ts:160]: Leaving this as it is — the early return is deliberate and predates this PR (`main` has the same `if (endpointRef.value == null) return;` at the top of `updateSnappedPosition`, and its `watch` returns early on `null` too), so the behaviour is unchanged here.

An endpoint goes `undefined` in one place only: `getSnapTarget()` in `FancyArrow.vue` returns `undefined` while the slide is not active and we are not printing, because the other slides may not be rendered and there is nothing to measure. Clearing the position there would drop the arrow the moment its slide stops being the active one — while that slide is still on screen animating out — and it would have to be drawn again from scratch on the way back. Keeping the last resolved position means the arrow stays put until `watch(isSlideActive)` and the 100ms poll re-measure it.

The position is cleared where clearing is right: when the SVG is gone (`rootElement == null`) or the snap target resolves to no measurable box.

---
_Generated by [Claude Code](https://claude.ai/code)_
